onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.h

```diff
@@ -4,10 +4,12 @@
 #pragma once
 
 #include <memory>
+#include <map>
 
 #include "core/framework/allocatormgr.h"
 #include "core/framework/execution_provider.h"
 #include "core/graph/graph_transformer.h"
+#include "mkldnn.hpp"
```

Comment:
forward declare mkldnn:memory instead of including mkldnn header. otherwise you leak mkldnn header to those that include mkldnn_execution_provider.h
did you test python bindings by building with --enable_pybind? the build there fails due to this issue.

Reply:
forward declared `mkldnn::memory`

```
#ifndef MKLDNN_HPP
namespace mkldnn {
class memory;
};
#endif
```